src/internal/communication.ts

```diff
@@ -0,0 +1,343 @@
+import { validOriginRegExp, version } from './constants';
+import { GlobalVars } from './globalVars';
+import { Handlers } from './handlers';
+import { MessageResponse, MessageRequest, ExtendedWindow, DOMMessageEvent } from './interfaces';
+
+export class Communication {
```

Comment:
I'm curious why you opted to create a static class here. This means that anywhere we want to use the members of this class, you need to import the class and call `Communication.foo`, which seems to be a pretty common pattern in this PR. It might be cleaner if these were free functions and variables on the module itself, so that in the places where it's used you could choose to do either of the following:
`import * as Communication from './communication'`
`import { sendMessageToParent } from './communication'`

The former would import everything from this file, and you'd still reference the members like `Communication.sendMessageToParent` as you're already doing - but the latter would mean that you could just call `sendMessageToParent`. As it stands now there isn't that flexibility, and you must import the entire Communication class from this module anywhere it's used.

Reply:
My idea was to copy the "style" of GlobalVars. There would be no problem having free functions, but I wonder what @ydogandjiev and @sachingupta think about this

Reply:
I also like using free functions, similar to ensureInitialized(). GlobalVars were use to keep non functional object. I think we should update static class to free functions for communications and handlers files